src/django/pfb_analysis/views.py

```diff
@@ -84,6 +87,48 @@ def results(self, request, pk=None):
             return Response(None, status=status.HTTP_404_NOT_FOUND)
 
 
+class AnalysisBatchViewSet(ViewSet):
+
+    @parser_classes([parsers.MultiPartParser])
+    def create(self, request, *args, **kwargs):
+        """ Trigger a new analysis batch given a well-formatted shapefile
+
+        Upload file to the 'file' key in a multipart form.
+
+        Each polygon/multipolygon feature in the shapefile will have a neighborhood created
+        for it if it doesn't exist, and the job for each neighborhood will immediately be submitted.
+
+        Each feature in the shapefile should have a "city" and "state" attribute. The "city"
+        attribute maps to Neighborhood.label and "state" maps to Neighborhood.state_abbrev.
+
+        If the "city" and "state" of an uploaded feature matches an existing neighborhood,
+        the existing one will be used and its geom updated with the one in the upload.
+
+        """
+        file_obj = request.data['file']
+        tmpdir = tempfile.mkdtemp()
+        try:
+            # Awkward. create_from_shapefile opens with fiona, so we can't just directly pass
+            #   in the file-like upload object. First we have to write it to disk, then pass
+            #   the on-disk location to the create method instead.
+            # Alternatively we could manually write to disk here, unzip and pass the fiona handle
+            #   to create_from_shapefile, but that duplicates tested logic we do anyways within
+            #   that method.
+            upload_filename = os.path.join(tmpdir, 'upload.zip')
+            with open(upload_filename, 'wb') as upload_file:
+                upload_file.write(file_obj.read())
+
+            batch = AnalysisBatch.objects.create_from_shapefile(upload_filename,
+                                                                submit=True,
+                                                                user=request.user)
```

Comment:
> The only work left to do here (if we have the time) would be to do a bit of error handling such that the API returns more helpful errors in specific cases. But, its going to be very difficult to pre-determine what those errors are likely to be. So, the current strategy of "log the entire error to the console" and teach how to view that seems like a good baseline to start from.

I think the whole error only gets returned if `settings.DEBUG` is true.  An easy alternate baseline would be catching `Exception as e` then raising `rest_framework.exceptions.APIException(e)` (i.e. give it the original exception, which will override the generic "There was an error" message).

That would still produce a 500 error, but without the stack trace (which I think is likely more confusing than helpful), and it would work the same in production as in development.

In terms of catching specific errors, I agree that it seems hard.  The one I generated was a
> "ValidationError: {'name': [u'This field cannot be blank.'], 'label': [u'This field cannot be blank.']}"

caused by removing the city name from one of the polygons in my shapefile.  That seems like a pretty plausible error for a user to make, but printing a nice error for it (describing it in a way that's general enough but not just vague, figuring out which geometry has a problem) would be tricky.

Reply:
This is a low cost change and would be valuable to implement as a first step effort.